### ISSUE TITLE
Allow explicitly overriding the automatic runner selection of `check-setup --set-default`

### ISSUE THREAD

**Issue by jameshadfield** [nextstrain/cli#110]
Currently `nextstrain check-setup --set-default` sets the default runner to the first runner which looks like it should work. Currently this list (in order) is docker, native, aws. I'd like this arg to take an optional argument so that the user can set their desired default runner, thus avoiding having to remember to specify it for each `nextrain build` command.

[E.g.](https://github.com/nextstrain/cli/blob/master/nextstrain/cli/command/check_setup.py#L105-L108)
```py
if opts.set_default:
    if opts.set_default==True: # might be opts.set_default=="", haven't tested
        config.set("core", "runner", supported_runners[0])
    elif opts.set_default in runner_status.keys():
        config.set("core", "runner", opts.set_default)
    else:
        print("Error...")
```

Apologies if this has been discussed elsewhere or is a deliberate design decision.

**Comment by tsibley** [comment]
Allowing the user to set a specific default runner is definitely part of the plan, e.g. with something like  `nextstrain config core.runner native` for example. The `config` command doesn't currently exist, but it's on the todo list and there's a draft PR that still needs some refinement.

I hadn't thought of allowing a value to `check-config --set-default`, though. This could work too, in addition to any more general `config` command. Since it's part of `check-setup`, what would the expected behaviour be if the requested default runner is known/valid but not detected as supported? I could see either:

1. Warn, but set as default.
2. Error, don't set as default.

I'd be inclined to (2) since this is part of `check-setup` and thus it would act as an assertion. (`config` could still be used to set it regardless of detected support.)

Thoughts?

In the meantime, you could avoid remembering to specify it every time by editing `~/.nextstrain/config` manually (if you're in a persistent environment) or with a command like:

```
echo -e '[core]\nrunner = native\n' >> ~/.nextstrain/config
```

Although this is not very user-friendly for instructions, which is what I suspect you're after…

(I also thought there was a `NEXTSTRAIN_RUNNER` environment variable which could be set, but I see that there's not.)

**Comment by jameshadfield** [comment]
`nextstrain config core.runner native` seems the best solution here, but I'm unsure of an ETA here (e.g. see https://github.com/nextstrain/cli/issues/68#issuecomment-614161668). While that's in progress, manual editing of `~/.nextstrain/config` would be ok - I forgot this was an option. I'll change this issue to ask for documentation of this. 

[Looking at the code](https://github.com/nextstrain/cli/search?q=config.get) we have only 4 (or so) config settings, most of which can be overridden by env variables. We should document these so that the people feel more comfortable understanding (& changing) config entries, as in my experience these these are sparsely documented for most software. I'll put this on my to-do list and try to write something over the coming weeks if no-one gets to it first. 

P.S. With regards to (1) vs (2) above, I would pick (1) but I can see how allowing a "check-support" command to pick a runner that it doesn't believe to be supported is slightly strange.

**Comment by tsibley** [comment]
Implemented `--set-default X` in #216, with the behaviour described by (1) above.

Documentation of all the config settings and runtimes is a larger and separate issue, IMO. The proposed `nextstrain config` is still desirable but is also a separate issue.

**Comment by tsibley** [comment]
I updated the issue title to match the original top post again.

**Comment by tsibley** [comment]
Resolved by [7dcf36a](https://github.com/nextstrain/cli/commit/7dcf36a19d4780f58bc8e7efd8c2ad2876d77b35).
